database/types/settings.go

```diff
@@ -0,0 +1,218 @@
+func (ps *Platform) Nullify() *Platform {
+	if ps == nil {
+		return nil
+	}
+
+	// check if the ID field should be false
+	if ps.ID.Int64 == 0 {
+		ps.ID.Valid = false
+	}
+
+	// check if the CloneImage field should be false
+	if len(ps.CloneImage.String) == 0 {
+		ps.CloneImage.Valid = false
+	}
+
+	// check if the CreatedAt field should be false
+	if ps.CreatedAt.Int64 < 0 {
+		ps.CreatedAt.Valid = false
+	}
+
+	// check if the UpdatedAt field should be false
+	if ps.UpdatedAt.Int64 < 0 {
+		ps.UpdatedAt.Valid = false
+	}
+
+	return ps
+}
+
+// ToAPI converts the Settings type
+// to an API Settings type.
+func (ps *Platform) ToAPI() *settings.Platform {
+	psAPI := new(settings.Platform)
+	psAPI.SetID(ps.ID.Int64)
+
+	psAPI.SetRepoAllowlist(ps.RepoAllowlist)
+	psAPI.SetScheduleAllowlist(ps.ScheduleAllowlist)
+
+	psAPI.Compiler = &settings.Compiler{}
+	psAPI.SetCloneImage(ps.CloneImage.String)
+	psAPI.SetTemplateDepth(int(ps.TemplateDepth.Int64))
+	psAPI.SetStarlarkExecLimit(uint64(ps.StarlarkExecLimit.Int64))
+
+	psAPI.Queue = &settings.Queue{}
+	psAPI.SetRoutes(ps.Routes)
+
+	psAPI.SetCreatedAt(ps.CreatedAt.Int64)
+	psAPI.SetUpdatedAt(ps.UpdatedAt.Int64)
+	psAPI.SetUpdatedBy(ps.UpdatedBy.String)
+
+	return psAPI
+}
+
+// Validate verifies the necessary fields for
+// the Settings type are populated correctly.
+func (ps *Platform) Validate() error {
+	// verify the CloneImage field is populated
+	if len(ps.CloneImage.String) == 0 {
+		return ErrEmptyCloneImage
+	}
+
+	// verify compiler settings are within limits
+	if ps.TemplateDepth.Int64 <= 0 {
+		return fmt.Errorf("template depth must be greater than zero, got: %d", ps.TemplateDepth.Int64)
+	}
+
+	if ps.StarlarkExecLimit.Int64 <= 0 {
+		return fmt.Errorf("starlark exec limit must be greater than zero, got: %d", ps.StarlarkExecLimit.Int64)
+	}
+
+	// ensure that all Settings string fields
+	// that can be returned as JSON are sanitized
+	// to avoid unsafe HTML content
+	ps.CloneImage = sql.NullString{String: util.Sanitize(ps.CloneImage.String), Valid: ps.CloneImage.Valid}
+
+	// ensure that all Queue.Routes are sanitized
+	// to avoid unsafe HTML content
+	for i, v := range ps.Routes {
+		ps.Routes[i] = util.Sanitize(v)
+	}
+
+	// ensure that all RepoAllowlist are sanitized
+	// to avoid unsafe HTML content
+	for i, v := range ps.RepoAllowlist {
+		ps.RepoAllowlist[i] = util.Sanitize(v)
+	}
+
+	// ensure that all ScheduleAllowlist are sanitized
+	// to avoid unsafe HTML content
+	for i, v := range ps.ScheduleAllowlist {
+		ps.ScheduleAllowlist[i] = util.Sanitize(v)
+	}
+
+	if ps.CreatedAt.Int64 < 0 {
+		return fmt.Errorf("created_at must be greater than zero, got: %d", ps.CreatedAt.Int64)
+	}
+
+	if ps.UpdatedAt.Int64 < 0 {
+		return fmt.Errorf("updated_at must be greater than zero, got: %d", ps.UpdatedAt.Int64)
+	}
+
+	return nil
+}
+
+// FromAPI converts the API Settings type
+// to a database Settings type.
+func FromAPI(s *settings.Platform) *Platform {
```

Comment:
```suggestion
func SettingsFromAPI(s *settings.Platform) *Platform {
```
Sorry just caught this. Need to have the resource part of the name if they are all in the `types` package.

Reply:
cool cool lemme tweak it

Reply:
Will address in follow up PR

Reply:
https://github.com/go-vela/server/pull/1126 🙇